src/cpp/multisolver_interface/SolverAbstract.cpp

```diff
@@ -0,0 +1,141 @@
+#include "antares-xpansion/multisolver_interface/SolverAbstract.h"
+
+namespace
+{
+bool verifyProblemDimensions(const SolverAbstract* merged, const SolverAbstract* master)
+{
+    return (merged->get_ncols() == master->get_ncols())
+           && (merged->get_nrows() == master->get_nrows())
+           && (merged->get_nelems() == master->get_nelems());
+}
+
+bool verifyObjectiveFunction(const SolverAbstract* merged, const SolverAbstract* master)
+{
+    std::vector<double> obj_merged(merged->get_ncols());
+    std::vector<double> obj_master(master->get_ncols());
+    merged->get_obj(obj_merged.data(), 0, merged->get_ncols() - 1);
+    master->get_obj(obj_master.data(), 0, master->get_ncols() - 1);
+
+    return std::ranges::equal(obj_merged, obj_master);
+}
+
+bool verifySolution(const SolverAbstract* merged, const SolverAbstract* master)
```

Comment:
I agree with @flomnes comment earlier, verifying solution is not needed, comparing equality of problems is what we really aim for

Reply:
I have removed the function